### PR TITLE
test: use a fake certificate to test revocation

### PR DESCRIPTION
**Pull request by phiz71**
**Description**

use a fake certificate to test revocation. Add also some documentation for the next time the certificate should be regenerated.

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.3.2`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-jws/1.3.2/gravitee-policy-jws-1.3.2.zip)

### CODE REVIEW

File: src/test/java/io/gravitee/policy/jws/JWSPolicyTest.java [hunk 1]
```diff
@@ -240,13 +240,20 @@ public void shouldValidateCRL() throws Exception {
...
         String fileContent = new String(Files.readAllBytes(Paths.get(pemPath)), Charset.forName(StandardCharsets.UTF_8.name()));
         X509Certificate cert = (X509Certificate) certificateFactory.generateCertificate(
             new ByteArrayInputStream(fileContent.getBytes(StandardCharsets.UTF_8))
         );
-        // WARNING: this test is based on real certificate from Wikipedia. Depending on the state of the expired certificates entries, it could be broken.
-        jwsPolicy.validateCRLSFromCertificate(cert, new BigInteger("18135611436097276018781887707242686781"));
+        /*
+         * WARNING:
+         * this test is based on a fake certificate from https://badssl.com/. (See also https://github.com/chromium/badssl.com)
+         * As this certificate will expire, it will need to be regenerated. Here is how to do it:
```
Comment (marcambier) [inline src/test/java/io/gravitee/policy/jws/JWSPolicyTest.java:252]: When this fake certificate expires, the test won't fail as it throws the same Exception, so we won't be to noticed to regenerate it.
Maybe we could catch the `CertificateException` and check the error message ?

Could we also get a dedicated `expired certificate` test case ?

Comment (phiz71) [reply src/test/java/io/gravitee/policy/jws/JWSPolicyTest.java:252]: - I replaced the *revoked.pem* file with the right one (it was not well generated)
 - I added an expiration test in the revocation unit test to be sure that the revoked certificate is not expired.

**Review by gaetanmaisse — APPROVED** [review]

File: src/test/java/io/gravitee/policy/jws/JWSPolicyTest.java (hunk 1 above)
Comment (marcambier) [reply src/test/java/io/gravitee/policy/jws/JWSPolicyTest.java:252]: great, thanks

**Review by marcambier — APPROVED** [review]

**Pull request merged**